hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/recon/TestReconTasks.java

```diff
@@ -74,21 +80,22 @@ public void init() throws Exception {
 
     conf.set("ozone.scm.stale.node.interval", "6s");
     conf.set("ozone.scm.dead.node.interval", "8s");
-    cluster =  MiniOzoneCluster.newBuilder(conf).setNumDatanodes(1)
+    cluster =  MiniOzoneCluster.newBuilder(conf).setNumDatanodes(3)
```

Comment:
> With 1 datanode it was working fine in local but was getting stuck in a wait condition in CI. After multiple such tries I found the CI working when we switch to 3 datanodes.
> 
> The performance after changes can be viewed in https://github.com/raju-balpande/apache_ozone/actions/runs/8091976415/job/22112283975 <img alt="image" width="1249" src="https://private-user-images.githubusercontent.com/146973984/309765869-9dbe2dcc-b6db-460e-95e3-29c1c8ea96bf.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk3Mzc4MjAsIm5iZiI6MTcwOTczNzUyMCwicGF0aCI6Ii8xNDY5NzM5ODQvMzA5NzY1ODY5LTlkYmUyZGNjLWI2ZGItNDYwZS05NWUzLTI5YzFjOGVhOTZiZi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzA2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMwNlQxNTA1MjBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kOTE1NTkyZTY0ZjdjM2NhNjBiYmNhNzMzZmEyY2RkMmZlYmE4ZWZiZGYzNjJmNGE3Njc1M2JmYjRkODYzNzA0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.hHlillOHvUTu-YEAwl-Q9-xT7GapZitgfqTxhmze6xk">
> 
> and previously it was https://github.com/raju-balpande/apache_ozone/actions/runs/7843423779/job/21404234425 <img alt="image" width="1236" src="https://private-user-images.githubusercontent.com/146973984/309765409-c51f2f93-5f27-4fee-b5c0-2ef2b174d09c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk3Mzc4MjAsIm5iZiI6MTcwOTczNzUyMCwicGF0aCI6Ii8xNDY5NzM5ODQvMzA5NzY1NDA5LWM1MWYyZjkzLTVmMjctNGZlZS1iNWMwLTJlZjJiMTc0ZDA5Yy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzA2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMwNlQxNTA1MjBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04OTNmYjIyMjkwZjU0MTE1MWFhMzhjOGU2NzY1NmRlODU5YTUyMTk5MGJjYjgwYWMxZWI5OTMyZTExYTc3OWYxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.FnJ-vdp0AYudkzTiF6cA70pZqTgqeK_x832l9_jbupE">

Can you tell which test case and which wait condition it was getting stuck when 1 DN was used for cluster ?

Reply:
Hi @devmadhuu ,
It was getting stuck on wait condition at TestReconTasks.java:173
`    LambdaTestUtils.await(120000, 6000, () -> {
      List<UnhealthyContainers> allMissingContainers =
          reconContainerManager.getContainerSchemaManager()
              .getUnhealthyContainers(
                  ContainerSchemaDefinition.UnHealthyContainerStates.MISSING,
                  0, 1000);
      return (allMissingContainers.size() == 1);
    });`
As I see in log https://github.com/raju-balpande/apache_ozone/actions/runs/7916623862/job/21611614999 
`Error:  Tests run: 3, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 365.519 s <<< FAILURE! - in org.apache.hadoop.ozone.recon.TestReconTasks
Error:  org.apache.hadoop.ozone.recon.TestReconTasks.testMissingContainerDownNode  Time elapsed: 300.006 s  <<< ERROR!
java.util.concurrent.TimeoutException: testMissingContainerDownNode() timed out after 300 seconds
	at java.util.ArrayList.forEach(ArrayList.java:1259)
	at java.util.ArrayList.forEach(ArrayList.java:1259)
	Suppressed: java.lang.InterruptedException: sleep interrupted
		at java.lang.Thread.sleep(Native Method)
		at org.apache.ozone.test.LambdaTestUtils.await(LambdaTestUtils.java:133)
		at org.apache.ozone.test.LambdaTestUtils.await(LambdaTestUtils.java:180)
		at org.apache.hadoop.ozone.recon.TestReconTasks.testMissingContainerDownNode(TestReconTasks.java:173)
		at java.lang.reflect.Method.invoke(Method.java:498)`

Reply:
Ok, thanks for the update @raju-balpande , However I am not sure why above test condition should timeout with just 1 DN in CI and pass in local, because with 1 DN in cluster, we are shutting down that only DN and so missing container count should be 1.